### PR TITLE
[test] Enable block splitting in MLPerf large file size nightly test

### PR DESCRIPTION
**Pull request by stephanie-wang**
## Why are these changes needed?

Enables block splitting in MLPerf train test for large file sizes. Upped the largest file size to 8192 images, which can OOM without block splitting.

### CODE REVIEW

File: release/air_tests/air_benchmarks/mlperf-train/file_size_benchmark.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 set -x
 
-NUM_IMAGES_PER_FILE="32 512 2048"
+NUM_IMAGES_PER_FILE="32 512 8192"
```
Comment (cadedaniel) [inline release/air_tests/air_benchmarks/mlperf-train/file_size_benchmark.sh:6]: FYI we'll likely have to rerun tests once Clarence's changes are in https://github.com/ray-project/ray/pull/29679/files#diff-61b7d0ee907414af24c086ed52e41d3896eddff5b0807b2eaa46a683205db486R6

**Comment by stephanie-wang** [comment]
Verified the test passes with block splitting on, results in worker OOMs when block splitting is off.

Final release test run [here](https://buildkite.com/ray-project/release-tests-pr/builds/20331).

**Comment by cadedaniel** [comment]
Did you see any perf regression when this was on vs. off? I'm trying to make sense of the numbers reported in your run:
```
[
    {
        "perf_metric_name": "ray.data_32-images-per-file_throughput-img-per-second",
        "perf_metric_value": 149.68785626367702,
        "perf_metric_type": "THROUGHPUT"
    },
    {
        "perf_metric_name": "ray.data_512-images-per-file_throughput-img-per-second",
        "perf_metric_value": 148.57266272968224,
        "perf_metric_type": "THROUGHPUT"
    },
    {
        "perf_metric_name": "ray.data_8192-images-per-file_throughput-img-per-second",
        "perf_metric_value": 138.0057545881724,
        "perf_metric_type": "THROUGHPUT"
    }
]
```

Screen Shot 2022-11-04 at 3 46 24 PM

--

Oh I think I know the reason -- I bet it's spilling since the num_images_per_epoch is calculated as max(images_per_file) * min_parallel = 8192*10, which is greater than the object store size.

Should we separate out these tests, or change this logic? Since we're not manually profiling anymore, we can probably relax the constraint that they all run with the same num_images_per_epoch, and thus avoid this spilling.

File: release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py [hunk 2]
```diff
@@ -481,15 +481,26 @@ def append_to_test_output_json(path, metrics):
...
+            ctx = ray.data.context.DatasetContext.get_current()
+            ctx.block_splitting_enabled = True
+
             datasets["train"] = build_dataset(
                 args.data_root,
                 args.num_images_per_epoch,
                 args.num_images_per_input_file,
             )
+            # Set a lower batch size for images to prevent OOM.
+            batch_size = 32
```
Comment (cadedaniel) [inline release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py:495]: Can we specify this from the top-level script? IIRC we specify it as an arg already?

**Comment by cadedaniel** [comment]
> Should we separate out these tests, or change this logic? Since we're not manually profiling anymore, we can probably relax the constraint that they all run with the same num_images_per_epoch, and thus avoid this spilling.

You can leave this to me and merge without fixing this. I was planning on fixing this to also fix https://github.com/ray-project/ray/issues/29912

**Review by cadedaniel — APPROVED** [review]
Please move the batch_size assignment to the top-level script, otherwise LGTM. Nice work, this is a cool improvement :)

File: release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py (hunk 2 above)
Comment (stephanie-wang) [reply release/air_tests/air_benchmarks/mlperf-train/resnet50_ray_air.py:495]: Addressed offline - this batch size is unrelated to the one used by the trainer.

**Pull request merged**